### DISCUSSION TITLE
When I have multiple lines of text, how should the line height be calculated?

### DISCUSSION THREAD

**Discussion by jmu201621143028** [harfbuzz/uharfbuzz#215] [Q&A] [answered]
Thank you for your great work.
i have a question. When I have multiple lines of text, how should the line height be calculated?

This is how I calculate it now. [ref](https://github.com/harfbuzz/uharfbuzz/pull/190/files)
```python
line_height = extents.ascender - extents.descender + extents.line_gap
cursor_x, cursor_y = -300, 0
texts = 'i love you\n我爱你\n愛してます'

"""
some code
"""

for pos, c in zip(positions, texts):
    if c == "\n":
        cursor_x = -300
        cursor_y += line_height * scale
        continue
    outline = get_glyph_outline(font_path, c)
    draw_outline_with_turtle(outline, scale=scale,
                             offset=(cursor_x + pos.x_offset * scale, cursor_y + pos.y_offset * scale))
    cursor_x += pos.x_advance * scale
    cursor_y += pos.y_advance * scale

```
Is there any better way to calculate **line_height**？

**Comment by khaledhosny** [comment, marked as answer]
That is as good as it gets.

**Reply by khaledhosny** [reply]
You should split your text into lines (at `\n`), then call `hb.shape()` on each line. Also, your code is not using the glyph returned from `hb.shape()`, so you will get the wrong outline if there is any glyph substitution happening.
